### PR TITLE
RAJA Benchmark fixes

### PR DESCRIPTION
**Pull request by drreynolds**
This PR fixes multiple errors in the RAJA-based advection_reaction_3D benchmark program, and moves the corresponding RAJA-specific source codes into a subfolder, in anticipation of an accompanying kokkos version.

### CODE REVIEW

File: benchmarks/advection_reaction_3D/raja/advection_reaction_3D.hpp [hunk 1]
```diff
@@ -124,6 +118,14 @@ struct UserData
   /* constructor that takes the context */
   UserData(SUNContext ctx) : ctx(ctx) {
     SUNContext_GetProfiler(ctx, &prof);
+    umask = nullptr;
```
Comment (balos1) [inline benchmarks/advection_reaction_3D/raja/advection_reaction_3D.hpp:121]: Initialize these in the initializer list instead.

**Review by balos1 — CHANGES_REQUESTED** [review]
Looks good. Just one minor requested change.

File: benchmarks/advection_reaction_3D/raja/advection_reaction_3D.cpp [hunk 2]
```diff
@@ -159,182 +154,121 @@ UserData::~UserData()
...
-          {
-            Bview(i,j,0) = Yview(i,j,nzl-1,0);
-            Bview(i,j,1) = Yview(i,j,nzl-1,1);
-            Bview(i,j,2) = Yview(i,j,nzl-1,2);
-          }
-
-        });
-      });
+
+    /* Create 3D views of send buffers */
```
Comment (gardner48) [inline benchmarks/advection_reaction_3D/raja/advection_reaction_3D.cpp:202]: Should this check the values of `NDIMS` for which buffers need to be packed?

File: benchmarks/advection_reaction_3D/raja/rhs3D.hpp [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+  SUNDIALS_MARK_END(udata->prof, "Neighbor Exchange");
+
+
+  /* compute advection at process boundaries */
+  if (c > 0.0)
+  {
+    /* Flow moving in the positive x,y,z direction:
+    *  boundaries are west face, south face, back face */
+
+    /*   Create 3D views of receive buffers */
```
Comment (gardner48) [inline benchmarks/advection_reaction_3D/raja/rhs3D.hpp:139]: Do we need to check `NDIMS` here also?

**Review by gardner48 — CHANGES_REQUESTED** [review]
I think some minor fixes are needed for `NDIMS != 3` but otherwise this looks good

File: benchmarks/advection_reaction_3D/raja/advection_reaction_3D.cpp (hunk 2 above)
Comment (drreynolds) [reply benchmarks/advection_reaction_3D/raja/advection_reaction_3D.cpp:202]: Good catch.  This one is an easy fix.

File: benchmarks/advection_reaction_3D/raja/rhs3D.hpp (hunk 3 above)
Comment (drreynolds) [reply benchmarks/advection_reaction_3D/raja/rhs3D.hpp:139]: Unfortunately, this is a more difficult fix.  If the flow is less than 3D, some parts of each kernel must be omitted, either resulting in (a) two `if` statements within each iteration of the forall (likely killing performance), or (b) separate 1D, 2D and 3D copies of each kernel (making things uglier and less maintainable).

I'll go with the latter of the two.

**Comment by drreynolds** [comment]
@gardner48: I just updated both `advection_reaction_3D.cpp` and `rhs3D.hpp` to [hopefully] support both 1D and 2D configurations.  I'll need to test these out thoroughly tomorrow.  If you guys could hold off on approving/merging until I report back, I'd appreciate it.

**Comment by drreynolds** [comment]
This passes all tests locally, and should ow be ready for review/approval.

**Review by gardner48 — APPROVED** [review]

**Review by balos1 — APPROVED** [review]

**Pull request merged**